### DISCUSSION TITLE
How to use iteration node?

### DISCUSSION THREAD

**Discussion by octadion** [langgenius/dify#4830] [Help]
how to use iteration node? from what i tried i am confused because input variable can't be used in input, does input only accept output from function?

**Comment by patryk20120** [comment]
Output of `Iteration` block is an array, so it cannot be used everywhere. It can be used only in blocks that accepts arrays e.g. `Code` block

**Comment by computersrmyfriends** [comment]
But, how does the iteration node work? Can it iterate over a json response from an LLM on the previous node? Any documentation on how to iterate?

**Comment by patryk20120** [comment]
Haven't found an documentation yet about Iterator block but you can iterate over an array of object that is an input to that block. The workflow inside will execute for every array saving the output to an array of results.

So if you would like to iterate over an array from http response you will need to do request using HTTP block, later extract the values from the response with Code block and then use Iterator. 

If you still will have an issue I can prepare some example import file

**Reply by octadion** [reply]
Could you please provide an example? It would be very helpful. Thank you!

**Reply by patryk20120** [reply]
Here you can find an example

[test chatbot.yml.txt](https://github.com/user-attachments/files/15526240/test.chatbot.yml.txt)
